File: Remonty/wells-back/src/main/java/com/finist/wellsback/models/Remont.java
package com.finist.wellsback.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.sql.Time;
import java.sql.Date;

@Entity                                                                                                                 //класс типа Entity указывает на класс, который на абстрактном уровне связан с таблицей в базе данных. Каждый объект, созданный этим классом, указывает на кортеж самой таблицы, содержащий информацию последней.
@Setter                                                                                                                 // генерирует set для каждого аргумента
@Getter                                                                                                                 // генерирует get для каждого аргумента
@AllArgsConstructor                                                                                                     // генерирует метод со всеми аргументами
@NoArgsConstructor                                                                                                      // генерирует пустой метод

@Table(name = "remonty")
public class Remont {

    @Id
    @GeneratedValue                                                                                                     //автоматическая генерация значения
    private Long id;

    @Column(name = "skvajina")                                                                                          //@Column используется для добавления имени столбца в таблицу конкретной базы данных
    private String skvajina;


    @Column(name = "kyst")                                                                                              //@Column используется для добавления имени столбца в таблицу конкретной базы данных
    private String kyst;


    @Column(name = "mestorojdenie")                                                                                     //@Column используется для добавления имени столбца в таблицу конкретной базы данных
    private String mestorojdenie;

    @ManyToOne(optional=true, fetch = FetchType.EAGER, cascade=CascadeType.REFRESH)                                     //@ManyToOne определяет однозначную ассоциацию с другим классом сущностей, который имеет кратность «многие к одному».
    @JoinColumn(name = " vid_remonta", nullable = true)
    private Vid vid_remonta;

    @Column(name = "prichina")                                                                                          //@Column используется для добавления имени столбца в таблицу конкретной базы данных
    private String prichina;


    @Temporal(TemporalType.DATE)                                                                                        //аннотация @Temporal  решает одну из основных проблем преобразования значений даты и времени из объекта Java в совместимый тип базы данных и их возврат в приложение.
    private Date date_start;

    @Temporal(TemporalType.TIME)                                                                                        //аннотация @Temporal  решает одну из основных проблем преобразования значений даты и времени из объекта Java в совместимый тип базы данных и их возврат в приложение.
    private Time time_start;


    @Temporal(TemporalType.DATE)                                                                                        //аннотация @Temporal  решает одну из основных проблем преобразования значений даты и времени из объекта Java в совместимый тип базы данных и их возврат в приложение.
    private Date date_end;


    @Temporal(TemporalType.TIME)                                                                                        //аннотация @Temporal  решает одну из основных проблем преобразования значений даты и времени из объекта Java в совместимый тип базы данных и их возврат в приложение.
    private Time time_end;

    @Column(name = "brigada")                                                                                           //@Column используется для добавления имени столбца в таблицу конкретной базы данных
    private String brigada;

    @Column(name = "primechanie")                                                                                       //@Column используется для добавления имени столбца в таблицу конкретной базы данных
    private String primechanie;
}

File: Remonty/wells-front/src/app/components/remonty/remont/Remont.ts
export class Remont {
    private _id: bigint;
    private _skvajina: string;
    private _kyst: string;
    private _mestorojdenie: string;
    private _vid_remonta: Object;
    private _prichina: string;
    private _date_start: string;
    private _time_start: string;
    private _date_end: string;
    private _time_end: string;
    private _brigada: string;
    private _primechanie: string;


    constructor(id: bigint, skvajina: string, kyst: string, mestorojdenie: string, vid_remonta: Object, prichina: string,
    date_start: string, time_start: string, date_end: string, time_end: string, brigada: string, primechanie: string){
        this._id = id;
        this._skvajina = skvajina;
        this._kyst = kyst;
        this._mestorojdenie = mestorojdenie;
        this._vid_remonta = vid_remonta;
        this._prichina = prichina;
        this._date_start = date_start;
        this._time_start = time_start;
        this._date_end = date_end;
        this._time_end = time_end;
        this._brigada = brigada;
        this._primechanie = primechanie;
    }

    get id(): bigint {
        return this._id;
    }

    set id(value: bigint){
        this._id = value;
    }

    get skvajina(): string {
        return this._skvajina;
    }

    set skvajina(value: string){
        this._skvajina = value;
    }

    get kyst(): string {
        return this._kyst;
    }

    set kyst(value: string){
        this._kyst = value;
    }

    get mestorojdenie(): string {
        return this._mestorojdenie;
    }

    set mestorojdenie(value: string){
      this._mestorojdenie = value;
    }

    get vid_remonta(): Object {
        return this._vid_remonta;
    }

    set vid_remonta(value: Object){
        this._vid_remonta = value;
    }

    get prichina(): string {
        return this._prichina;
    }

    set prichina(value: string){
      this._prichina = value;
    }

    get date_start(): string {
        return this._date_start;
    }

    set date_start(value: string){
      this._date_start = value;
    }

    get time_start(): string {
        return this._time_start;
    }

    set time_start(value: string){
        this._time_start = value;
    }

    get date_end(): string {
        return this._date_end;
    }

    set date_end(value: string){
        this._date_end = value;
    }

    get time_end(): string {
      return this._time_end;
    }

    set time_end(value: string){
        this._time_end = value;
    }

    get brigada(): string {
        return this._brigada;
    }

    set brigada(value: string){
        this._brigada = value;
    }

    get primechanie(): string {
        return this._primechanie;
    }

    set primechanie(value: string){
        this._primechanie = value;
    }
}
